### PR TITLE
pack: systemd unit parameterizing support

### LINKED ISSUE #723: Add support for parametrizing systemd units during pack
`cartridge-cli` uses "systemd-unit-params.yml" file to add parameters to the generated systemd unit. Need to support the same in `tt`. Add an option to use custom file name for parameters.

### PR DESCRIPTION
**Pull request by psergee**
@TarantoolBot document
Title: tt pack: systemd unit parameterizing support

The patch adds support for parameterizing systemd unit files allowing a user to change `LimitNOFile` and set environment variables in unit files. Unit parameters can be defined in using new `--unit-params-file` command line option or by creating a `systemd-unit-params.yml` file in the application directory. The `systemd-unit-params.yml` file has a higher priority than the command line option and overrides it.

Parameters file example:
```
FdLimit: 200
instance-env:
  INSTANCE: "inst:%i"
  TARANTOOL_WORKDIR: "/tmp"
```

Pack command example:
`tt pack rpm --unit-params-file <params file>`

Closes #723

### CODE REVIEW

File: cli/pack/systemd_dir.go [hunk 1]
```diff
@@ -77,54 +76,71 @@ func initSystemdDir(packCtx *PackCtx, opts *config.CliOpts,
...
 	}
 }
 
+type systemdUnitParams struct {
+	AppName     string         `yaml:"AppName"`
+	ExecArgs    string         `yaml:"ExecArgs"`
+	TT          string         `yaml:"TT"`
+	ConfigPath  string         `yaml:"ConfigPath"`
+	FdLimit     uint64         `yaml:"FdLimit"`
+	InstanceEnv map[string]any `yaml:"instance-env"`
```
Comment (oleg-jukovec) [inline cli/pack/systemd_dir.go:95]: It may work, but I don't sure.
```suggestion
	InstanceEnv map[string]string `yaml:"instance-env"`
```

**Review by oleg-jukovec — APPROVED** [review]

File: cli/pack/systemd_dir_test.go [hunk 2]
```diff
@@ -182,15 +168,34 @@ func Test_initSystemdDir(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name: "Default template and partly environment params defined",
```
Comment (DerekBum) [inline cli/pack/systemd_dir_test.go:172]:
```suggestion
			name: "Default template and partially defined environment params",
```

**Review by DerekBum — APPROVED** [review]

File: cli/pack/systemd_dir.go (hunk 1 above)
Comment (psergee) [reply cli/pack/systemd_dir.go]: Done.

File: cli/pack/systemd_dir_test.go (hunk 2 above)
Comment (psergee) [reply cli/pack/systemd_dir_test.go]: Fixed.

**Pull request merged**